### PR TITLE
python3Packages.jsonfield: init at 3.1.0

### PR DESCRIPTION
**Pull request by MrMebelMan**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+buildPythonPackage rec {
+  pname = "jsonfield";
+  version = "3.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0yl828cd0m8jsyr4di6hcjdqmi31ijh5vk57mbpfl7p2gmcq8kky";
+  };
+
+  checkInputs = [ pytest pytest-django ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:12]:
```suggestion
  checkInputs = [ pytestCheckHook pytest-django ];
```

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchPypi, buildPythonPackage, django, pytest, pytest-django }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:1]:
```suggestion
{ stdenv, fetchPypi, buildPythonPackage, django, pytestCheckHook , pytest-django }:
```

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    inherit pname version;
+    sha256 = "0yl828cd0m8jsyr4di6hcjdqmi31ijh5vk57mbpfl7p2gmcq8kky";
+  };
+
+  checkInputs = [ pytest pytest-django ];
+
+  checkPhase = ''
+    PYTHONPATH="$(pwd):$PYTHONPATH" \
+    DJANGO_SETTINGS_MODULE=tests.settings \
+    pytest #tests/tests.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:17]: We can't use pytestCheckHook here, right?

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0yl828cd0m8jsyr4di6hcjdqmi31ijh5vk57mbpfl7p2gmcq8kky";
+  };
+
+  checkInputs = [ pytest pytest-django ];
+
+  checkPhase = ''
+    PYTHONPATH="$(pwd):$PYTHONPATH" \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:15]: Does it any harm to export these?

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  propagatedBuildInputs = [ django ];
+
+  meta = with stdenv.lib; {
+    description = "jsonfield is a reusable model field that allows you to store validated JSON, automatically handling serialization to and from the database.";
+    homepage = "https://github.com/rpkilby/jsonfield/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ mrmebelman ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:29]: One final new line to much.

File: pkgs/development/python-modules/jsonfield/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  checkPhase = ''
+    PYTHONPATH="$(pwd):$PYTHONPATH" \
+    DJANGO_SETTINGS_MODULE=tests.settings \
+    pytest #tests/tests.py
+  '';
+
+  propagatedBuildInputs = [ django ];
+
+  meta = with stdenv.lib; {
+    description = "jsonfield is a reusable model field that allows you to store validated JSON, automatically handling serialization to and from the database.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jsonfield/default.nix:23]:
```suggestion
    description = "Reusable model field that allows you to store validated JSON, automatically handling serialization to and from the database";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107536` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python37Packages.jsonfield</li>
    <li>python38Packages.jsonfield</li>
    <li>python39Packages.jsonfield</li>
  </ul>
</details>

```
  File "/nix/store/kq4r0xn7ia8alvv0v1bp0dac0mvnjddf-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/nix/store/kq4r0xn7ia8alvv0v1bp0dac0mvnjddf-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/nix/store/kq4r0xn7ia8alvv0v1bp0dac0mvnjddf-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/nix/store/4qm4lh6iv2pkrsh8lmwkjjvqvr3ivv84-python3.8-pytest-django-4.1.0/lib/python3.8/site-packages/pytest_django/plugin.py", line 320, in pytest_load_initial_conftests
    dj_settings.DATABASES
  File "/nix/store/72qx63lh6h0l56izan1lvc4px0mqrsdp-python3.8-Django-2.2.17/lib/python3.8/site-packages/django/conf/__init__.py", line 79, in __getattr__
    self._setup(name)
  File "/nix/store/72qx63lh6h0l56izan1lvc4px0mqrsdp-python3.8-Django-2.2.17/lib/python3.8/site-packages/django/conf/__init__.py", line 66, in _setup
    self._wrapped = Settings(settings_module)
  File "/nix/store/72qx63lh6h0l56izan1lvc4px0mqrsdp-python3.8-Django-2.2.17/lib/python3.8/site-packages/django/conf/__init__.py", line 188, in __init__
    if zoneinfo_root.exists() and not zone_info_file.exists():
  File "/nix/store/v3qi2n1cb9bj8h3076p3rz5fg0kpzz53-python3-3.8.6/lib/python3.8/pathlib.py", line 1391, in exists
    self.stat()
  File "/nix/store/v3qi2n1cb9bj8h3076p3rz5fg0kpzz53-python3-3.8.6/lib/python3.8/pathlib.py", line 1197, in stat
    return self._accessor.stat(self)
PermissionError: [Errno 1] Operation not permitted: '/usr/share/zoneinfo'
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107536` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.jsonfield</li>
    <li>python38Packages.jsonfield</li>
    <li>python39Packages.jsonfield</li>
  </ul>
</details>

File: pkgs/development/python-modules/jsonfield/default.nix (hunk 4 above)
Comment (MrMebelMan) [reply pkgs/development/python-modules/jsonfield/default.nix:15]: I think I've found a way to use pytestCheckHook by exporting just the DJANGO_SETTING_MODULE in the preCheck phase

File: pkgs/development/python-modules/jsonfield/default.nix (hunk 3 above)
Comment (MrMebelMan) [reply pkgs/development/python-modules/jsonfield/default.nix:17]: I haven't figured out how, but I guess I'll better just not customize this phase.

File: pkgs/development/python-modules/jsonfield/default.nix (hunk 4 above)
Comment (MrMebelMan) [reply pkgs/development/python-modules/jsonfield/default.nix]: https://github.com/NixOS/nixpkgs/pull/107536/commits/3ef93692b5a360be8575e242bf833fdc15472a55

**Comment by SuperSandro2000** [comment]
Can you squash the commit into one init commit?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107536` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.jsonfield</li>
    <li>python38Packages.jsonfield</li>
    <li>python39Packages.jsonfield</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 7]
```diff
@@ -3174,6 +3174,8 @@ in {
 
   jsondate = callPackage ../development/python-modules/jsondate { };
 
+  jsonfield = callPackage ../development/python-modules/jsonfield { };
+
   jsondiff = callPackage ../development/python-modules/jsondiff { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3179]:
```suggestion
  jsondiff = callPackage ../development/python-modules/jsondiff { };

  jsonfield = callPackage ../development/python-modules/jsonfield { };
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 107536` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.jsonfield</li>
    <li>python38Packages.jsonfield</li>
    <li>python39Packages.jsonfield</li>
  </ul>
</details>

**Comment by MrMebelMan** [comment]
Reordered jsondiff and jsonfield and squashed into init commit

**Pull request merged**